builtin/common/class.lua

```diff
@@ -0,0 +1,119 @@
+class = {}
+
+function class.new(name, ...)
+	local cls = {}
+
+	-- Copy inherited classes in reverse order to make the first ones overwrite later ones
+	local bases = {...}
+	for i = #bases, 1, -1 do
+		for k, v in pairs(bases[i]) do
+			cls[class.copy(k)] = class.copy(v)
+		end
+	end
+
+	cls.__name = name
```

Comment:
Why? This becomes problematic if there is `scope_a.class` and `scope_b.class`: both will have the same name.